### PR TITLE
Auto detect package versions

### PR DESCRIPTION
**Pull request by peterfeiner**
Automatically detect the right versions of openjdk and libsnappy. This obviates the --openjdk_version flag. Also makes hadoop installation on ubuntu-16.04 work.

### CODE REVIEW

**Comment by peterfeiner** [comment]
@tedsta Please take a look.

File: perfkitbenchmarker/linux_packages/hadoop.py [hunk 1]
```diff
@@ -69,7 +69,10 @@ def YumInstall(vm):
 
 def AptInstall(vm):
   """Installs Hadoop on the VM."""
-  vm.InstallPackages('libsnappy1 libsnappy-dev')
+  libsnappy = 'libsnappy1'
+  if not vm.HasPackage(libsnappy):
+    libsnappy = 'libsnappy1v5'
```
Comment (tedsta) [inline perfkitbenchmarker/linux_packages/hadoop.py]: libsnappy1v5 is the package name on Ubuntu 16.04? A comment there would be nice.

File: perfkitbenchmarker/linux_packages/openjdk.py [hunk 2]
```diff
@@ -21,17 +21,26 @@
...
+flags.DEFINE_string('openjdk_version', None, 'Version of openjdk to use. '
+                    'By default, the version of openjdk is automatically '
+                    'detected.')
+
+
+def _OpenJdkPackage(vm, format_string):
+  version = FLAGS.openjdk_version
+  if version is None:
+    if vm.HasPackage(format_string.format('7')):
+      version = '7'
```
Comment (tedsta) [inline perfkitbenchmarker/linux_packages/openjdk.py]: We prefer version 7 over version 8?

File: perfkitbenchmarker/virtual_machine.py [hunk 3]
```diff
@@ -374,6 +374,16 @@ def RemoteCommand(self, command, should_log=False, ignore_failure=False,
     """
     raise NotImplementedError()
 
+  def TryRemoteCommand(self, command, **kwargs):
```
Comment (tedsta) [inline perfkitbenchmarker/virtual_machine.py]: Nice! How very DRY of you :)

Comment (peterfeiner) [inline perfkitbenchmarker/virtual_machine.py]: Err, I don't get the joke :confused:

Comment (tedsta) [inline perfkitbenchmarker/virtual_machine.py]: Not really a joke - DRY = Don't Repeat Yourself

deduplicated code

File: perfkitbenchmarker/linux_packages/openjdk.py (hunk 2 above)
Comment (tedsta) [inline perfkitbenchmarker/linux_packages/openjdk.py]: Makes sense

**Pull request merged**